src/StackExchange.Redis/ResultProcessor.cs

```diff
@@ -221,7 +221,7 @@ public virtual bool SetResult(PhysicalConnection connection, Message message, in
             {
                 try
                 {
-                    logging.Log?.WriteLine($"Response from {bridge?.Name} / {message.CommandAndKey}: {result}");
+                    logging.Log?.LogInfo($"Response from {bridge?.Name} / {message.CommandAndKey}: {result}");
```

Comment:
Apart from using stateful callback APIs that avoid allocations, providing Logging API with pre-interpolated strings effectively disables structured logging (https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#log-message-template or aka.ms/r9 (internal only) and search structured logging). 

There's a push (internally, at least in some orgs) towards structured logging so if it was possible to support that, it'd be doubly awesome :).

Reply:
Ofc, the change 'd need to be made in LogProxy and threaded through all layers.

Reply:
This is true, but it's also more expensive at every call site (due to the allocation of the `object[]` array at the end). Ideally we could enable structure logging without the overhead :-/ I'm not against structured logging - tons of uses, but it's odd to me that logging is such a performance hog across so much and there isn't an easier path that's also efficient here. Hoping we continue to see that improve in the coming .NET releases if we can push a bit on logging APIs. `ILogger`'s shape makes that hard though, when realizing wins through the pipe.

Reply:
It's more expensive at every call site _when the logging actually goes trough_, i.e. for levels that are enabled. But for log-calls that don't end up being used (loglevel, ...), you don't pay for interpolating the strict, which I'd guess would be roughly similar to the cost or higher than allocating object[] array.

Reply:
If you have a logger _at all_, it's almost certainly to go through here. We're already null checking against that, so we've solved the first problem and moved on to the second. In our case, because of the elvis approach, it' universally more expensive :)

Reply:
Ah right, brain-fart on my side 😅. The array needs to be allocated always. In that case I see how it's a performance negative.

Reply:
Do the Source Generated logger messages help here? I admit I haven't looked at them properly. https://learn.microsoft.com/en-us/dotnet/core/extensions/logger-message-generator